### ISSUE TITLE
needless_borrow: suggest removing several references at a time.

### ISSUE THREAD

**Issue by disconnect3d** [rust-lang/rust-clippy#5327]
Hey,

It would be nice if Clippy could detect needless references as in here:
```rust
fn main() {
    let st = String::from("zomg");
    myfunc(&&&&&st);
}

fn myfunc(param: &str) {
    println!("param: {}", param);
}
```

The `myfunc(&&&&&st);` should be `myfunc(&st);`.

Side note: there is a similar, needless lifetime parameter detecting, that would trigger in e.g. `fn myfunc<'a>(param: &'a str)`

**Comment by matthiaskrgr** [comment]
Hi, this is [already implemented](https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow) but you have to opt-in manually (`#[warn(clippy::needless_borrow)]
`)

````rust
#[warn(clippy::needless_borrow)]
fn main() {
    let st = String::from("zomg");
    myfunc(&&&&&st);
}

fn myfunc(param: &str) {
    println!("param: {}", param);
}

````
should make clippy warn about it. :)

**Comment by disconnect3d** [comment]
@matthiaskrgr is there a way to opt in with a CLI argument? Also, how do I list all linters/detectors?

**Comment by disconnect3d** [comment]
@matthiaskrgr Also, the suggestion given is not the right one and applying it and re-running `cargo clippy` will still warn:

```
➜  refs git:(master) ✗ cargo clippy
    Checking refs v0.1.0 (/Users/dc/playground/rust-2/refs)
warning: this expression borrows a reference that is immediately dereferenced by the compiler
 --> src/main.rs:6:12
  |
6 |     myfunc(&&&&&st);
  |            ^^^^^^^ help: change this to: `&&&&st`
  |
note: lint level defined here
 --> src/main.rs:1:8
  |
1 | #[warn(clippy::needless_borrow)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow

    Finished dev [unoptimized + debuginfo] target(s) in 0.12s
```

seems like something to be improved

**Comment by matthiaskrgr** [comment]
`cargo clippy -- -Wclippy::needless-borrow` should work.

Mhh, looks like the link only looks at one reference at a time.. :thinking:

**Comment by flip1995** [comment]
All lints are listed here: https://rust-lang.github.io/rust-clippy/

Just choose the version of Clippy you're using and it will show you all lints in this version with their documentation. (master ~= nightly)
